### PR TITLE
Async device open and create from device, passphrase entry, requires #5355

### PR DESCRIPTION
**Pull request by ph4r05**
- Requires https://github.com/monero-project/monero-gui/pull/2019
- Passphrase entry on host added, requires early listener setting https://github.com/monero-project/monero/pull/5355
- Wallet open and create from device shows splash to indicate a possible long process
- Create from a device is async to support passphrase entry

Screenshot 2019-03-27 at 09 27 35
Screenshot 2019-03-27 at 09 27 48
Screenshot 2019-03-27 at 09 28 22

## Building

In order to build this PR for testing you need to fetch my original branch

```bash
git clone --recursive https://github.com/monero-project/monero-gui.git
cd monero-gui
git remote add ph4r05 https://github.com/ph4r05/monero-gui.git
git fetch --all
git checkout -b trezor-pass ph4r05/trezor-pass

cd monero
git remote add ph4r05 https://github.com/ph4r05/monero.git
git fetch --all
git checkout -b trezor/api_pass ph4r05/trezor/api_pass
git submodule update --recursive
cd -
./build.sh
```

## Troubleshooting

If you experience `Bad offset calculation` error when creating a transaction the remote node you are using is probably running too old version.

The code is known to work with version `131076`. For testing purposes, try node `173.255.205.142:18089`, it should work smoothly.

### CODE REVIEW

File: components/PassphraseDialog.qml [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+                Layout.fillWidth: true
+
+                font.pixelSize: 16 * scaleRatio
+                font.family: MoneroComponents.Style.fontLight.name
+
+                color: MoneroComponents.Style.defaultFontColor
+            }
+
+            Label {
+                text: qsTr("Warning: passphrase entry on host is a security risk as it can be captured by mallware. It is advised to prefer device-based passphrase entry.");
```
Comment (xmrdsc) [inline components/PassphraseDialog.qml]: malware

File: components/PassphraseDialog.qml [hunk 2]
```diff
@@ -0,0 +1,327 @@
+// Copyright (c) 2017, The Monero Project
```
Comment (xmrdsc) [inline components/PassphraseDialog.qml]: `> // Copyright (c) 2014-2019, The Monero Project`

File: components/PassphraseDialog.qml (hunk 1 above)
Comment (ph4r05) [reply components/PassphraseDialog.qml]: 😓

**Comment by xmrdsc** [comment]
Nice PR. I have one small complaint with the QML, that is, `components/PassphraseDialog.qml` seems to be based off `components/PasswordDialog.qml`. They share similar functionality. I can attempt to get that into one file to avoid code duplication - and you wont have to deal with QML ;)

**Comment by ph4r05** [comment]
> Nice PR. I have one small complaint with the QML, that is, `components/PassphraseDialog.qml` seems to be based off `components/PasswordDialog.qml`. They share similar functionality. I can attempt to get that into one file to avoid code duplication - and you wont have to deal with QML ;)

Thanks! 

The dialog is based on `NewPasswordDialog.qml`. We need to have 2 text boxes for passphrase entry as it is not checked by Trezor (on purpose, changes seed, plausible deniability, different wallets).

Moreover we need to have a warning there so user knows that host-based entry should be only fallback.

If you can factor out common code that would be nice. I didn't want to make huge changes to existing QMLs so this seemed to me as a path of minimal resistance :)

**Comment by xmrdsc** [comment]
Ill look at it soon^tm, great work.

**Comment by selsta** [comment]
Before this patch, it would display the following if the Ledger version was outdated:
Screenshot 2019-04-03 at 23 13 21
After this change, it shows “Creating wallet from device” for a short time and then fails without any error.

While the logs show the following, it would be great if this behaviour could be like before.
```
2019-04-03 21:13:50.097	  0x70000b235000	ERROR	device.ledger	src/device/device_ledger.cpp:322	Unsupported device application version: 1.2.2 At least 1.3.1 is required.
2019-04-03 21:13:50.098	     0x1135755c0	WARNING	frontend	src/wallet/api/wallet.cpp:403	qrc:/wizard/WizardController.qml:420: Error: Cannot assign [undefined] to QString
```

**Comment by ph4r05** [comment]
I've added "Proceed to the device" splash logic to make it more explicit that some user action on the device is required. Otherwise user can just wait for operation to complete not knowing Trezor wants a confirmation.

**Comment by ph4r05** [comment]
> the following, it would be great if this behaviour could be like before.

Ah thanks for info! I will check that

**Comment by ph4r05** [comment]
@selsta could you pls try again?

**Comment by selsta** [comment]
@ph4r05 Works now :)

**Comment by xmrdsc** [comment]
Few days ago I argued we should avoid having to include `PassphraseDialog.qml` in this PR, however, I think we (GUI) should re-do password prompts anyway, possibly make it a dedicated screen as per @GBKS design (on zpl.io).

Current PR is fine by me.

**Pull request merged**